I've rebased but now we have several integration tests failures in the docker observer:
```
--- FAIL: TestObserverEmitsEndpointsIntegration (2.19s)
    integration_test.go:53: 
        	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/extension/observer/dockerobserver/integration_test.go:53
        	Error:      	Expected nil, but got: &fmt.wrapError{msg:"http: invalid Host header, host port waiting failed: could not start container: creating reaper failed: failed to create container", err:(*fmt.wrapError)(0xc0003df7a0)}
        	Test:       	TestObserverEmitsEndpointsIntegration
```

--- a/extension/observer/dockerobserver/integration_test.go
+++ b/extension/observer/dockerobserver/integration_test.go
@@ -36,6 +36,7 @@ func (h *testHost) ReportFatalError(err error) {
 var _ component.Host = (*testHost)(nil)
 
 func TestObserverEmitsEndpointsIntegration(t *testing.T) {
+	t.Skip("See https://github.com/testcontainers/testcontainers-go/issues/1359")
 	image := "docker.io/library/nginx"
 	tag := "1.17"
 
@@ -82,6 +83,7 @@ func TestObserverEmitsEndpointsIntegration(t *testing.T) {
 }
 
 func TestObserverUpdatesEndpointsIntegration(t *testing.T) {
+	t.Skip("See https://github.com/testcontainers/testcontainers-go/issues/1359")
 	image := "docker.io/library/nginx"
 	tag := "1.17"
 
@@ -143,6 +145,7 @@ func TestObserverUpdatesEndpointsIntegration(t *testing.T) {
 }
 
 func TestObserverRemovesEndpointsIntegration(t *testing.T) {
+	t.Skip("See https://github.com/testcontainers/testcontainers-go/issues/1359")
 	image := "docker.io/library/nginx"
 	tag := "1.17"
 
@@ -185,6 +188,7 @@ func TestObserverRemovesEndpointsIntegration(t *testing.T) {
 }
 
 func TestObserverExcludesImagesIntegration(t *testing.T) {
+	t.Skip("See https://github.com/testcontainers/testcontainers-go/issues/1359")
 	ctx := context.Background()
 	req := testcontainers.ContainerRequest{
 		Image:        "docker.io/library/nginx:1.17",

--- a/internal/coreinternal/scraperinttest/scraperint.go
+++ b/internal/coreinternal/scraperinttest/scraperint.go
@@ -164,6 +164,7 @@ func (it *IntegrationTest) createNetwork(t *testing.T) testcontainers.Network {
 }
 
 func (it *IntegrationTest) createContainers(t *testing.T) *ContainerInfo {
+	t.Skip("See https://github.com/testcontainers/testcontainers-go/issues/1359")
 	var wg sync.WaitGroup
 	ci := &ContainerInfo{
 		containers: make(map[string]testcontainers.Container, len(it.containerRequests)),

--- a/internal/coreinternal/scraperinttest/scraperint.go
+++ b/internal/coreinternal/scraperinttest/scraperint.go
@@ -143,6 +143,7 @@ func (it *IntegrationTest) Run(t *testing.T) {
 }
 
 func (it *IntegrationTest) createNetwork(t *testing.T) testcontainers.Network {
+	t.Skip("See https://github.com/testcontainers/testcontainers-go/issues/1359")
 	var errs error
 
 	var network testcontainers.Network

--- a/receiver/dockerstatsreceiver/integration_test.go
+++ b/receiver/dockerstatsreceiver/integration_test.go
@@ -53,6 +53,7 @@ func paramsAndContext(t *testing.T) (rcvr.CreateSettings, context.Context, conte
 }
 
 func createNginxContainer(ctx context.Context, t *testing.T) testcontainers.Container {
+	t.Skip("See https://github.com/testcontainers/testcontainers-go/issues/1359")
 	req := testcontainers.ContainerRequest{
 		Image:        "docker.io/library/nginx:1.17",
 		ExposedPorts: []string{"80/tcp"},

--- a/receiver/dockerstatsreceiver/integration_test.go
+++ b/receiver/dockerstatsreceiver/integration_test.go
@@ -86,6 +86,8 @@ func hasResourceScopeMetrics(containerID string, metrics []pmetric.Metrics) bool
 
 func TestDefaultMetricsIntegration(t *testing.T) {
 	t.Parallel()
+	// remove nolint when https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/24240 is resolved
+	// nolint:staticcheck
 	params, ctx, cancel := paramsAndContext(t)
 	defer cancel()
 
@@ -140,6 +142,8 @@ func TestMonitoringAddedAndRemovedContainerIntegration(t *testing.T) {
 
 func TestExcludedImageProducesNoMetricsIntegration(t *testing.T) {
 	t.Parallel()
+	// remove nolint when https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/24240 is resolved
+	// nolint:staticcheck
 	params, ctx, cancel := paramsAndContext(t)
 	defer cancel()
 

--- a/receiver/snmpreceiver/integration_test.go
+++ b/receiver/snmpreceiver/integration_test.go
@@ -94,6 +94,7 @@ var (
 )
 
 func getContainer(t *testing.T, req testcontainers.ContainerRequest) testcontainers.Container {
+	t.Skip("See https://github.com/testcontainers/testcontainers-go/issues/1359")
 	require.NoError(t, req.Validate())
 	container, err := testcontainers.GenericContainer(
 		context.Background(),

--- a/receiver/sqlqueryreceiver/integration_test.go
+++ b/receiver/sqlqueryreceiver/integration_test.go
@@ -248,6 +248,7 @@ func TestPostgresIntegrationLogsTrackingWithStorage(t *testing.T) {
 }
 
 func startPostgresDbContainer(t *testing.T, externalPort string) testcontainers.Container {
+	t.Skip("See https://github.com/testcontainers/testcontainers-go/issues/1359")
 	req := testcontainers.ContainerRequest{
 		Image: "postgres:9.6.24",
 		Env: map[string]string{

